sound/soc/sof/ipc4-loader.c

```diff
@@ -256,26 +254,111 @@ static int sof_ipc4_load_library_by_uuid(struct snd_sof_dev *sdev,
 	if (ret)
 		goto release;
 
-	ret = xa_insert(&ipc4_data->fw_lib_xa, lib_id, fw_lib, GFP_KERNEL);
+	ret = xa_insert(&ipc4_data->fw_lib_xa, *lib_id, fw_lib, GFP_KERNEL);
 	if (unlikely(ret))
 		goto release;
 
-	kfree(fw_filename);
-
+	(*lib_id)++;
```

Comment:
what sense does it make incrementing this if the caller then ignores it?

Reply:
I think there were a reason why I did this, but cannot recall, I guess the increment can be done in a caller function.
Likely I had a different implementation which I reworked to arrive to this one..